src/executor/stack/executor.rs

```diff
@@ -831,15 +831,24 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet>
 		// Avoid uncontrolled `u64` casting
 		let refunded_gas =
 			u64::try_from(self.state.metadata().gasometer.refunded_gas()).unwrap_or_default();
-		self.state.metadata().gasometer.total_used_gas()
+		let total_used_gas = self.state.metadata().gasometer.total_used_gas();
+		let total_used_gas_refunded = self.state.metadata().gasometer.total_used_gas()
 			- min(
-				self.state.metadata().gasometer.total_used_gas() / self.config.max_refund_quotient,
+				total_used_gas / self.config.max_refund_quotient,
 				refunded_gas,
-			)
+			);
+		// EIP-7623: max(total_used_gas, floor_gas)
+		if self.config.has_floor_gas
+			&& total_used_gas_refunded < self.state.metadata().gasometer.floor_gas()
+		{
+			self.state.metadata().gasometer.floor_gas()
+		} else {
+			total_used_gas_refunded
+		}
 	}
 
 	/// Get fee needed for the current executor, given the price.
-	pub fn fee(&self, price: U256) -> U256 {
+	pub fn fee(&mut self, price: U256) -> U256 {
```

Comment:
Why does this need to be mutable now?

Reply:
Good catch! After new tests `peactra-devnet-5-1.2.0` some logic was changed. I cleared `mut` in other places. And that line just forget. Funny that clippy didn't warn it.

Reply:
Fixed: da311299776dd381df23798ff5148cb066961980